api/Dockerfile

```
@@ -29,21 +29,46 @@ COPY pyproject.toml requirements.lock ./
+FROM python:3.14-slim@sha256:7a500125bc50693f2214e842a621440a1b1b9cbb2188f74ab045d29ed2ea5856 AS node-deps
+
+WORKDIR /app
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    nodejs \
+    npm \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install app compiler Node.js dependencies
+COPY api/src/services/app_compiler/package.json api/src/services/app_compiler/package-lock.json* /app/src/services/app_compiler/
+RUN cd /app/src/services/app_compiler && \
```

Hadolint (2.14.0)

[warning] Use WORKDIR to switch to a directory

(DL3003)